data/src/test/java/org/oppia/android/data/backends/gae/RemoteAuthNetworkInterceptorTest.kt

```diff
@@ -0,0 +1,207 @@
+@LooperMode(LooperMode.Mode.PAUSED)
+class RemoteAuthNetworkInterceptorTest {
+
+  @Inject
+  lateinit var remoteAuthNetworkInterceptor: RemoteAuthNetworkInterceptor
+
+  @Inject
+  lateinit var context: Context
+
+  private lateinit var retrofit: Retrofit
+
+  lateinit var topicService: TopicService
+
+  private val testVersionName = "1.0"
+
+  private val testVersionCode = 1
+
+  private val topicName = "Topic1"
+
+  private val mockWebServer = MockWebServer()
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+    setUpApplicationForContext()
+    setUpRetrofit()
+  }
+
+  private fun setUpApplicationForContext() {
+    val packageManager = Shadows.shadowOf(context.packageManager)
+    val applicationInfo =
+      ApplicationInfoBuilder.newBuilder()
+        .setPackageName(context.packageName)
+        .build()
+    val packageInfo =
+      PackageInfoBuilder.newBuilder()
+        .setPackageName(context.packageName)
+        .setApplicationInfo(applicationInfo)
+        .build()
+    packageInfo.versionName = testVersionName
+    packageInfo.versionCode = testVersionCode
+    packageManager.installPackage(packageInfo)
+  }
+
+  @Test
+  fun testNetworkInterceptor_withoutHeaders_addsCorrectHeaders() {
+    mockWebServer.enqueue(MockResponse().setBody("{}"))
+    val call = topicService.getTopicByName(topicName)
+    val serviceRequest = call.request()
+    assertThat(serviceRequest.header("api_key")).isNull()
+    assertThat(serviceRequest.header("app_package_name")).isNull()
+    assertThat(serviceRequest.header("app_version_name")).isNull()
+    assertThat(serviceRequest.header("app_version_code")).isNull()
+
+    call.execute()
+    val interceptedRequest = mockWebServer.takeRequest(timeout = 8, unit = TimeUnit.SECONDS)
+    verifyRequestHeaders(interceptedRequest!!.headers)
+  }
+
+  @Test
+  fun testNetworkInterceptor_withHeaders_setsCorrectHeaders() {
+    mockWebServer.enqueue(MockResponse().setBody("{}"))
+    val call = topicService.getTopicByName(topicName)
+
+    val serviceRequest = topicService.getTopicByName(topicName).request()
+    val request = serviceRequest.newBuilder()
+      .addHeader("api_key", "wrong_api_key")
+      .addHeader("app_package_name", "wrong_package_name")
+      .addHeader("app_version_name", "wrong_version_name")
+      .addHeader("app_version_code", "wrong_version_code")
+      .build()
+
+    call.execute()
+    val interceptedRequest = mockWebServer.takeRequest(timeout = 8, unit = TimeUnit.SECONDS)
+    verifyRequestHeaders(interceptedRequest!!.headers)
+  }
+
+  private fun verifyRequestHeaders(headers: Headers) {
+    assertThat(headers.get("api_key")).isEqualTo("test_api_key")
+    assertThat(headers.get("app_package_name")).isEqualTo(context.packageName)
+    assertThat(headers.get("app_version_name")).isEqualTo("1.0")
+    assertThat(headers.get("app_version_code")).isEqualTo("1")
+  }
+
+  private fun setUpRetrofit() {
+    mockWebServer.dispatcher
```

Comment:
This isn't doing anything--I suggest removing it.

Reply:
Ah oops thanks! Done